types/ioredis/index.d.ts

```diff
@@ -875,10 +875,20 @@ declare namespace IORedis {
 
     type NodeRole = 'master' | 'slave' | 'all';
 
+    type CallbackFunction<T = any> = (err?: NodeJS.ErrnoException | null, result?: T) => void;
+
     interface Cluster extends NodeJS.EventEmitter, Commander {
         connect(callback: () => void): Promise<any>;
         disconnect(): void;
         nodes(role?: NodeRole): Redis[];
+        quit(callback?: CallbackFunction<'OK'>): Promise<'OK'>;
```

Comment:
Doesn't the cluster implement more or less *all* methods the regular `Redis` interface has?
It should probably inherit those methods from a shared interface rather than copy/pasting it since it seems many more are missing.

Reply:
I haven't had a chance to fully go through all of the ioredis code, but it looks like you are correct. I am unsure as to how these are added right now, possibly through the [transaction module](https://github.com/luin/ioredis/blob/dff3fe9b2a3e9d9a260637e020e701a570c34fe4/lib/transaction.ts). I also see it importing commands from [redis-commands](https://github.com/luin/ioredis/blob/dff3fe9b2a3e9d9a260637e020e701a570c34fe4/lib/cluster/index.ts#L15), as well.

Reply:
After another quick look, it looks like there are a lot of shared methods between the Redis and Pipeline interfaces, as well. This could be a much bigger issue that originally thought.

All I wanted was to be able to use the Cluster interface correctly! Ha! 😸

Reply:
Yeah according to the code in ioredis it takes all but the monitor command from `redis-commands` and dynamically adds them all to the `Commander.prototype` which `Cluster` extends
https://github.com/luin/ioredis/blob/master/lib/commander.js#L59